compiler/qsc_partial_eval/tests/intrinsics.rs

```diff
@@ -642,3 +642,259 @@ fn call_to_intrinsic_mresetz_adds_callable_and_generates_instruction() {
                 Return"#]],
     );
 }
+
+#[test]
+fn call_to_intrinsic_begin_estimate_caching_with_classical_values_does_not_generate_instructions() {
+    let program = compile_and_partially_evaluate(indoc! {
+        r#"
+        namespace Test {
+            open Microsoft.Quantum.ResourceEstimation;
+            @EntryPoint()
+            operation Main() : Unit {
+                let _ = BeginEstimateCaching("test", 0);
+            }
+        }
+        "#,
+    });
+    assert_block_instructions(
+        &program,
+        BlockId(0),
+        &expect![[r#"
+        Block:
+            Call id(1), args( Integer(0), Pointer, )
+            Return"#]],
+    );
+}
+
+#[test]
+fn call_to_intrinsic_begin_estimate_caching_with_dynamic_values_does_not_generate_corresponding_call_instruction(
+) {
+    let program = compile_and_partially_evaluate(indoc! {
+        r#"
+        namespace Test {
+            open Microsoft.Quantum.ResourceEstimation;
+            open QIR.Intrinsic;
+            @EntryPoint()
+            operation Main() : Unit {
+                use q = Qubit();
+                let i = __quantum__qis__m__body(q) == Zero ? 0 | 1;
+                let _ = BeginEstimateCaching("test", i);
```

Comment:
Same thing here, an instruction inside an if-expression would confirm the control flow behavior.

Reply:
Updated test.